scheduler/src/main/java/org/pentaho/platform/scheduler2/ws/DefaultSchedulerService.java

```diff
@@ -99,7 +105,16 @@ private void updateJob( String jobId, Map<String, ParamValue> jobParams, JobTrig
   /** {@inheritDoc} */
   public Job[] getJobs() throws SchedulerException {
     IScheduler scheduler = PentahoSystem.get( IScheduler.class, "IScheduler2", null ); //$NON-NLS-1$
-    return scheduler.getJobs( null ).toArray( new Job[0] );
+    IPentahoSession session = PentahoSessionHolder.getSession();
+    String principalName = session.getName();
+    Boolean canAdminister = PentahoSystem.get( IAuthorizationPolicy.class ).isAllowed( ADMIN_PERM );
+
+    return scheduler.getJobs( job -> {
+      if ( canAdminister ) {
+        return !IBlockoutManager.BLOCK_OUT_JOB_NAME.equals( job.getJobName() );
+      }
+      return principalName.equals( job.getUserName() );
```

Comment:
is this safe for different casing for user names and all that madness (I recall a recent issue about that)

Reply:
@pamval The filter is the same as we have in PUC in Schedule perspective now and it works correctly. See https://github.com/pentaho/pentaho-platform/blob/8ad9067e790c6997bd8aaed5e48ba5ee12df1e76/extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java#L531. Pedro, what issue do you mean?

Reply:
Can't locate it right now, but it had to do with the fact that you could login in spoon as ADMIN (or Admin or aDMin or whatever) and then you could not do certain actions or view certain content because the owner of the content was the user admin (all lower case). It was a fairly recent issue, I believe. Can you run a very quick test for that case?

Reply:
@pamval Yes,sure. Will check it

Reply:
I checked, the case is ok. Logged in via Spoon using usernames with upper/lower letters -> the returned list of schedules is correct